tensorflow_addons/image/cutout_ops.py

```diff
@@ -0,0 +1,168 @@
+from tensorflow_addons.image.utils import from_4D_image, to_4D_image
+from tensorflow.python.keras.utils import conv_utils
+
+
+def _get_image_wh(images, data_format):
+    if tf.equal(tf.rank(images), 4):
+        if data_format == "channels_last":
+            image_height, image_width = tf.shape(images)[1], tf.shape(images)[2]
+        else:
+            image_height, image_width = tf.shape(images)[2], tf.shape(images)[3]
+    else:
+        image_height, image_width = tf.shape(images)[0], tf.shape(images)[1]
+
+    return image_height, image_width
+
+
+def _norm_params(images, mask_size, data_format):
+    mask_size = tf.convert_to_tensor(mask_size)
+    if tf.equal(tf.rank(mask_size), 0):
+        mask_size = tf.stack([mask_size, mask_size])
+    data_format = conv_utils.normalize_data_format(data_format)
+    image_height, image_width = _get_image_wh(images, data_format)
+    return mask_size, data_format, image_height, image_width
+
+
+def random_cutout(
+    images: TensorLike,
+    mask_size: TensorLike,
+    constant_values: Number = 0,
+    seed: Number = None,
+    data_format: str = "channels_last",
+) -> tf.Tensor:
+    """Apply cutout (https://arxiv.org/abs/1708.04552) to images.
+
+    This operation applies a (2*pad_size[0] x 2*pad_size[1]) mask of zeros to
+    a random location within `img`. The pixel values filled in will be of the
+    value `replace`. The located where the mask will be applied is randomly
+    chosen uniformly over the whole images.
+
+    Args:
+      images: A tensor of shape
+        (num_images, num_rows, num_columns, num_channels)
+        (NHWC), (num_images, num_channels, num_rows, num_columns)(NCHW), (num_rows, num_columns, num_channels) (HWC), or
+        (num_rows, num_columns) (HW).
+      mask_size: Specifies how big the zero mask that will be generated is that
+        is applied to the images. The mask will be of size
+        (2*pad_size[0] x 2*pad_size[1]).
+      constant_values: What pixel value to fill in the images in the area that has
+        the cutout mask applied to it.
+      seed: A Python integer. Used in combination with `tf.random.set_seed` to
+        create a reproducible sequence of tensors across multiple calls.
+      data_format: A string, one of `channels_last` (default) or `channels_first`.
+        The ordering of the dimensions in the inputs.
+        `channels_last` corresponds to inputs with shape
+        `(batch_size, ..., channels)` while `channels_first` corresponds to
+        inputs with shape `(batch_size, channels, ...)`.
+    Returns:
+      An image Tensor.
+    """
+    mask_size, data_format, image_height, image_width = _norm_params(
+        images, mask_size, data_format
+    )
+
+    cutout_center_height = tf.random.uniform(
+        shape=[], minval=0, maxval=image_height, dtype=tf.int32, seed=seed
+    )
+    cutout_center_width = tf.random.uniform(
+        shape=[], minval=0, maxval=image_width, dtype=tf.int32, seed=seed
```

Comment:
If I understand correctly, if a batch of images is passed and random cutout is applied, the mask will be at the same place for all the images in the batch, right? Is this something we want? Maybe users would expect (as an augmentation strategy) to have the mask being at different places for all the images in the batch. 
If the original implementation uses the same mask for all the images in the batch, let's go with this implementation, but the function shouldn't be public. I believe a public `random_cutout` shoud respect the principle of least astonishment. 
What do you think?

Reply:
Exactly! In a batch the cutout operation has to be random for each image.